sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -651,7 +651,7 @@ public void validate(PBegin input) {
         }
 
         final String executingProject = bqOptions.getProject();
-        if (query != null && !Strings.isNullOrEmpty(query.get())) {
+        if (query != null && (!query.isAccessible() || !Strings.isNullOrEmpty(query.get()))) {
```

Comment:
(even if it may not quite be an issue for some runners that swap out this Java implementation of BigQueryIO.)